packages/backend-common/src/database/sqlite3.ts

```diff
@@ -29,6 +31,20 @@ export function createSqliteDatabaseClient(
   overrides?: Knex.Config,
 ) {
   const knexConfig = buildSqliteDatabaseConfig(dbConfig, overrides);
+
+  // If storage on disk is used, ensure that the directory exists
+  if (
+    typeof knexConfig.connection === 'object' &&
+    (knexConfig.connection as Knex.Sqlite3ConnectionConfig).filename
+  ) {
+    const { filename } = knexConfig.connection as Knex.Sqlite3ConnectionConfig;
+    const directory = path.dirname(filename);
```

Comment:
oh so here we strip the last part - so the actual file name is always thrown away, and replaced with the dynamic names?

Reply:
No, this is just for creating the directory where the files are placed, we still use the filenames.

Reply:
I'm probably reading things wrong. But if the filename is `/a/b`, this code will make sure to create `/a` right? And then the other code appends the database name to form `/a/b/backstage_plugin_catalog.sqlite` right?

Reply:
The other way around: The `buildSqliteDatabaseConfig` takes `/a/b` and generates a path like `/a/b/backstage_plugin_catalog.sqlite`. But before opening the database, we extract `/a/b` from it, create that directory and then open `/a/b/backstage_plugin_catalog.sqlite`.